lib/internal/bootstrap/node.js

```diff
@@ -426,23 +426,23 @@
       emitAfter
     } = NativeModule.require('internal/async_hooks');
 
-    process._fatalException = function(er) {
+    process._fatalException = function(er, fromPromise) {
       // It's possible that defaultTriggerAsyncId was set for a constructor
       // call that threw and was never cleared. So clear it now.
       clearDefaultTriggerAsyncId();
 
       if (exceptionHandlerState.captureFn !== null) {
         exceptionHandlerState.captureFn(er);
-      } else if (!process.emit('uncaughtException', er)) {
-        // If someone handled it, then great.  otherwise, die in C++ land
+      } else if (fromPromise || !process.emit('uncaughtException', er)) {
+        // If someone handled it, then great. Otherwise, die in C++ land
```

Comment:
Random nit: can we remove the word `die` from here? That has been objections to this sort of terminology before in Node. 

(I still think we should call the `uncaughtException` hook here.